client/lib/pages/email_login.dart

```diff
@@ -68,52 +69,20 @@ class _EmailLogInState extends State<EmailLogIn> {
               ),
               Padding(
                 padding: EdgeInsets.all(20.0),
-                child: isLoading
-                    ? CircularProgressIndicator()
-                    : RaisedButton(
-                        color: Colors.lightBlue,
-                        onPressed: () {
-                          if (_formKey.currentState.validate()) {
-                            setState(() {
-                              isLoading = true;
-                            });
-                            logInToFb();
-                          }
-                        },
-                        child: Text('Submit'),
-                      ),
-              )
-            ]))));
-  }
-
-  void logInToFb() {
-    FirebaseAuth.instance
-        .signInWithEmailAndPassword(
-            email: emailController.text, password: passwordController.text)
-        .then((result) {
-      isLoading = false;
-      Navigator.pushReplacement(
-        context,
-        MaterialPageRoute(builder: (context) => IndexPage(uid: result.user.uid)),
-      );
-    }).catchError((err) {
-      print(err.message);
-      showDialog(
-          context: context,
-          builder: (BuildContext context) {
-            return AlertDialog(
-              title: Text("Error"),
-              content: Text(err.message),
-              actions: [
-                FlatButton(
-                  child: Text("Ok"),
+                child: RaisedButton(
+                  color: Colors.lightBlue,
                   onPressed: () {
-                    Navigator.of(context).pop();
+                    authService.signInWithMail(
+                        email: emailController.text,
+                        password: passwordController.text);
+
+                    Navigator.of(context).pushAndRemoveUntil(
```

Comment:
Maybe we should send them to a loading page instead of going back to the auth page here, it feels like a bug even though its just running a background process. Also how are we going to handle auth when the device is offline?

Reply:
AuthPage resolves the current auth state and redirects the users to a certain page. I am not sure if there is a better way as flutter currently doesn't support protected routes like other frameworks. Therefore, we are passing the user through the AuthPage every time the `authState` changes. Will try finding a better solution.